### ISSUE TITLE
plug-in support for nems + nems_db

### ISSUE THREAD

**Issue by svdavid** [LBHB/NEMS0#78]
Generalization of @bburan 's #22 . Goal is to make nems lab independent and move all lbhb_specific stuff to nems_lbhb, which I'm happy to store in nems_db for now. A few steps:

1. formalize keyword system in xforms helper. So I think we need a registry similar to modelspec keywords.
2. Add support to nems_config to point to custom scripts that will register non-standard keywords. eg, 
```nems_config.my_loader_keywords=['nems_lbhb.loader_keywords']
nems_config.my_fitter_keywords=['nems_lbhb.fitter_keywords']
nems_config.my_modelspec_keywords=['nems_lbhb.modelspec_keywords',...]```
3. These keywords can reference functions in any library. Happily, it looks like xforms is already smart enough to import those functions as they're needed.
4. Then new keywords can generate xforms commands the same way they're generated now.
5. Once this infrastructure is in place, we can migrate lbhb-specific stuff out of preprocessing.py, initializers.py, xforms_helper.py

**Comment by svdavid** [comment]
Working in the plugins branch.

**Reorganize config files to go in the corresponding main libary, eg:**
```
nems/config/settings.py  /defaults.py
nems_db/config/settings.py  /defaults.py
nems_web/config/settings.py  /defaults.py
```

**Plugins for loader:**
eg, `load_keyword = "ozgf100ch18"`
determine `key_head = <get first alpha string or string preceding a dash "-" --> "ozgf">`

then run `load_lib[key_head](load_keyword, recording_uri)`
where
```
load_lib = {'ozgf': nems.xfhelper.ozgf,
   'env': nems.xfhelper.env,
   ... etc.}
```
Return value is a an xfspec.

Plugin libraries would then add new parsing functions to load_lib.

**Plugins for fitter:**
same deal but without `recording_uri` parameter?

**Comment by jacobpennington** [comment]
Resolved with merge of #120
